### ISSUE TITLE
Order of initializing metrics stream and circuitBreaker instance

### ISSUE THREAD

**Issue by yury-kozlov** [nodeshift/opossum#336]
Hi, is it mandatory to create instance of circuitBreaker before starting hystrix metrics stream ?
When I open the stream, and then create dynamically circuit breakers, I can't see them on hystrix dashboard (get error "Unable to connect to Command Metric Stream."), however stream can be accessed from browser and I can see reported events in real time).

If I define circuit breakers ahead and then open the stream afterwards, everything works fine.

## Operating System: Windows

Thanks

**Comment by lance** [comment]
@yury-kozlov hi - it seems as though you have answered your own question. Honestly, I think it should not matter what order they are created in. You haven't provided any of your code, so I can only guess what you are doing. I just created a little test that seems to have no problem.

```js
'use strict';

const test = require('tape');
const cb = require('../');
const { passFail } = require('./common');

test('It should not matter that the stream is accessed first', t => {
  t.plan(2);
  const stream = cb.stats;
  const circuitOne = cb(passFail, {
    rollingCountTimeout: 100,
    rollingCountBuckets: 1,
    name: 'circuit one'
  });
  const circuitTwo = cb(passFail, {
    rollingCountTimeout: 100,
    rollingCountBuckets: 1,
    name: 'circuit two'
  });
  let circuitOneStatsSeen = false;
  let circuitTwoStatsSeen = false;
  stream.on('data', blob => {
    const obj = JSON.parse(blob.substring(6));
    if (obj.name === 'circuit one') circuitOneStatsSeen = true;
    else if (obj.name === 'circuit two') circuitTwoStatsSeen = true;
  });
  circuitOne.fire(10).then(_ => circuitTwo.fire(10)).then(_ => {
    t.ok(circuitOneStatsSeen, 'circuit one stats seen');
    t.ok(circuitTwoStatsSeen, 'circuit two stats seen');
    t.end();
  });
});
```

**Comment by yury-kozlov** [comment]
Thanks for the example!
My problem was that on each call I created a **new** instance of circuit breaker and on completion, I disposed it (using **shutdown()** method), so there was not enough time for metrics stream to collect the statistics (in addition, this approach was wrong also because statistical data will be overwritten on each subsequent instance of circuit breaker).

Another conclusion that may be useful for others is that the order of initialization doesn't really matter, but hystrix dashboard will display circuits (and corresponding graphs) only when at least one circuit breaker is up (not shutdown). Otherwise it will show "Unable to connect to Command Metric Stream" message.
